examples/plugins/traffic-control/docker.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// DockerClient internal data structure
+type DockerClient struct {
+	store  *Store
+	client *docker.Client
+}
+
+// NewDockerClient instantiates a new DockerClient
+func NewDockerClient(store *Store) (*DockerClient, error) {
+	dc, err := docker.NewClient("unix:///var/run/docker.sock")
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to docker daemon: %v", err)
+	}
+	return &DockerClient{
+		store:  store,
+		client: dc,
+	}, nil
+}
+
+// Start docker client
+func (c *DockerClient) Start() {
+	for {
+		func() {
```

Comment:
If you replace the `return`s with `continue`s, and remove the defer (by moving the cleanup to after the for loop), we wouldn't need the lambda here.